pkg/util/node/node.go

```diff
@@ -122,7 +122,7 @@ func GetNodeIP(client clientset.Interface, hostname string) net.IP {
 		node, err := client.CoreV1().Nodes().Get(hostname, metav1.GetOptions{})
 		if err != nil {
 			klog.Errorf("Failed to retrieve node info: %v", err)
-			return false, nil
+			return false, err
```

Comment:
I think this is wrong.  It is called from `wait.ExponentialBackoff`, which will early-terminate in case of error.  So you are effectively changing the "log the error and retry" into "log the error and abort".

Reply:
Do we need to decide to continue to try again or just fail immediately according to the returned error of `Get`? 🤔For example, give `client` is not authorized to access node resource.

Reply:
We don't check the error code here, so your change would only alter the retry behavior. As @thockin mentioned, this is not what the code intends to do.